### PR TITLE
Query instance id in gobi transformer instead of isbn

### PR DESCRIPTION
**Pull request by jgreben**
The gobi transformer was incorrectly querying isbn. Now queries instance id. Also refactors the loops so that it only queries for holdings once per marc record and items once per holding.

### CODE REVIEW

File: libsys_airflow/plugins/data_exports/marc/gobi.py [hunk 1]
```diff
@@ -52,62 +71,47 @@ def generate_list(self, marc_file) -> pathlib.Path:
                         """
                         isbn = isbn.replace("-", "")
                         isbn = re.sub(r"\s.*", "", isbn)
-                        if not re.search(
+                        if re.search(
```
Comment (jermnelson) [inline libsys_airflow/plugins/data_exports/marc/gobi.py:74]: Maybe use a complied regex for ISBN before this `for` loop? (Realize that this isn't new code but more of a refactor to make the for loop more efficient)

**Review by jermnelson — APPROVED** [review]
Non-blocking suggestion to compile ISBN regex

File: libsys_airflow/plugins/data_exports/marc/gobi.py (hunk 1 above)
Comment (jgreben) [reply libsys_airflow/plugins/data_exports/marc/gobi.py]: :+1 just pushed a commit.

File: libsys_airflow/dags/data_exports/gobi_selections.py [hunk 2]
```diff
@@ -71,23 +67,6 @@
...
-    )
-
-    transform_marc_fields = PythonOperator(
-        task_id="transform_folio_remove_marc_fields",
-        python_callable=remove_fields_from_marc_files,
-        op_kwargs={
-            "marc_file_list": "{{ ti.xcom_pull('fetch_marc_records_from_folio') }}"
-        },
-    )
-
```
Comment (shelleydoljack) [inline libsys_airflow/dags/data_exports/gobi_selections.py:90]: Sure, I guess we don't need these if the generate_isbn_lists can use the untransformed MARC from SRS and lookup holdings/items info in folio.

File: libsys_airflow/plugins/data_exports/marc/gobi.py [hunk 3]
```diff
@@ -21,7 +21,8 @@ def gobi_list_from_marc_files(marc_file_list: str):
 
 
 class GobiTransformer(Transformer):
-    def generate_list(self, marc_file) -> pathlib.Path:
+    def generate_list(self, marc_file):
+
         # marc_path is data-export-files/gobi/marc-files/updates/YYYYMMDD.mrc
         marc_path = pathlib.Path(marc_file)
```
Comment (shelleydoljack) [inline libsys_airflow/plugins/data_exports/marc/gobi.py:27]: Is the marc file even used here anymore? Maybe, I guess if it is "untransformed" MARC, right?

**Review by shelleydoljack — APPROVED** [review]

File: libsys_airflow/plugins/data_exports/marc/gobi.py (hunk 3 above)
Comment (jgreben) [reply libsys_airflow/plugins/data_exports/marc/gobi.py:27]: Yes, the gobi transformer looks at the 020, 856, and 956 of the marc record, but not the 999s.

**Comment by jgreben** [comment]
@jermnelson @shelleydoljack I added a couple of commits to fix the archiving of gobi files: See: https://github.com/sul-dlss/libsys-airflow/pull/1033/commits/5772c700773a7e19057500fe494ac53d33c4b53f and https://github.com/sul-dlss/libsys-airflow/pull/1033/commits/42157708075d4f295a691b70bb33e1efff9a03ed

**Comment by shelleydoljack** [comment]
> @jermnelson @shelleydoljack I added a couple of commits to fix the archiving of gobi files: See: [5772c70](https://github.com/sul-dlss/libsys-airflow/commit/5772c700773a7e19057500fe494ac53d33c4b53f) and [4215770](https://github.com/sul-dlss/libsys-airflow/commit/42157708075d4f295a691b70bb33e1efff9a03ed)

But this will transmit the `*.mrc` files and not just `*.txt` files. We need to figure out a different way then to only transmit the txt files but then be able to archive all mrc and txt files in the gobi directory.

**Comment by shelleydoljack** [comment]
@jermnelson @jgreben see https://github.com/sul-dlss/libsys-airflow/pull/1038 as a solution to my comment on this PR.

**Pull request merged**